modules/delegates.js

```diff
@@ -183,51 +183,19 @@ __private.getDelegatesFromPreviousRound = function(cb, tx) {
 		});
 };
 
-/**
- * Generates delegate list and checks if block generator publicKey matches delegate id.
- *
- * @param {block} block
- * @param {function} source - Source function for get delegates
- * @param {function} cb - Callback function
- * @returns {setImmediateCallback} cb, err
- * @todo Add description for the params
- */
-__private.validateBlockSlot = function(block, source, cb) {
-	const round = slots.calcRound(block.height);
-	self.generateDelegateList(round, source, (err, activeDelegates) => {
-		if (err) {
-			return setImmediate(cb, err);
-		}
-
-		const currentSlot = slots.getSlotNumber(block.timestamp);
-		const delegateId = activeDelegates[currentSlot % ACTIVE_DELEGATES];
-
-		if (delegateId && block.generatorPublicKey === delegateId) {
-			return setImmediate(cb);
-		}
-		library.logger.error(
-			`Expected generator: ${delegateId} Received generator: ${
-				block.generatorPublicKey
-			}`
-		);
-		return setImmediate(cb, `Failed to verify slot: ${currentSlot}`);
-	});
-};
-
 /**
  * Gets the assigned delegate to current slot and returns its keypair if present.
  *
  * @private
  * @param {number} slot
- * @param {number} round
+ * @param {block} block
```

Comment:
This param expects block height, right?
If so, please rename it to `blockHeight` and type number instead.

Reply:
Good catch, thanks.